test/cases/base/load_test.rb

```diff
@@ -72,6 +73,20 @@ def setup
         {Time.at(1009839600) => "Ruby in a Nutshell"},
         {Time.at(1199142000) => "The Ruby Programming Language"}
     ]}
+
+    @complex_books = {
+      books: {
+        "Complex.String&-Character*|=_+()!~": {
+          isbn: 1009839690,
+          author: 'Frank Smith'
+        },
+        "16Candles": {
+          isbn: 1199142400,
+          author: 'John Hughes'
+        }
+      }
+    }
+
```

Comment:
_Should_ Active Resource support this? These are meant to map to valid Ruby constant names, so raising `NameError` is expected and desirable. Can you represent this as an array of books instead?

Reply:
You could represent this as an array of books and that would probably be ideal. Unfortunately its not always possible or practical to make sure the data is formatted the way you prefer. In the Refund Transaction example, the contents of the receipt object is not supplied by our application and comes from a payment gateway. We sanitize it and its fine to consume RESTfully as all the keys are valid JSON, just not by ActiveResource (at least not 100% of the time). 

Running into a NameError seems to have come up a few times in the past and should help developers who are fairly new to development and those with data not entirely under their control, or is hard to reformat. 

https://github.com/rails/activeresource/issues/33 (keys were java package names containing special characters)
https://github.com/rails/activeresource/issues/138 (keys contained a server port number)
https://github.com/rails/activeresource/issues/269 (looks like keys were numbers)

Reply:
Thanks for linking those. Agree we should be liberal in what we parse when we can.

Looking at these other cases, the resources are unknown only in the sense that we don't know which Ruby class to map them to. Perhaps "unnamed" or "anonymous" resource more clearly reflects their disposition?

Reply:
For sure. I was thinking this morning when refactoring that maybe unknown was the wrong word choice. I think UnnamedResource rolls off the tongue better, but I'm not particular.